content/en/apps/guides/hosting/4.x/app-developer.md

```diff
@@ -0,0 +1,144 @@
+---
+title: "App Developer Hosting in CHT 4.x"
+linkTitle: "App Developer Hosting"
+weight: 40
+aliases:
+- /apps/guides/hosting/app-developer
+description: >
+  Hosting the CHT when developing apps
+---
+
+{{% alert title="Note" %}} This guide assumes you are a CHT app developer wanting to either run concurrent instances of the CHT, or easily be able to switch between different instances without loosing any data while doing so. To do development on the CHT core itself, see the [development guide]({{< relref "apps/guides/hosting/core-developer.md" >}}).
+
+To deploy the CHT 3.x in production, see either [AWS hosting]({{< relref "apps/guides/hosting/3.x/self-hosting.md" >}}) or [Self hosting]({{< relref "apps/guides/hosting/3.x/ec2-setup-guide.md" >}}). 4.x production hosting guides are coming soon!{{% /alert %}}
+
+
+## Getting started
+
+Be sure to meet the [CHT hosting requirements]({{< relref "apps/guides/hosting/requirements" >}}) first. As well, if any other CHT 4.x instances using Docker Compose are running locally, stop them otherwise port, storage volume and container name conflicts may occur. To stop ALL containers, you can use `docker kill $(docker ps -q)`.
+
+After meeting these requirements, create a directory and download the developer YAML files in the directory you want to store them. This example uses `~/cht-4-app-developer` as the directory:
+
+```shell 
+mkdir  ~/cht-4-app-developer && cd ~/cht-4-app-developer
+curl -s -o docker-compose.yml https://raw.githubusercontent.com/medic/cht-upgrade-service/main/docker-compose.yml
+curl -s -o cht-core.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic%3Amedic%3Amaster/docker-compose/cht-core.yml
+curl -s -o cht-couchdb.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic%3Amedic%3Amaster/docker-compose/cht-couchdb.yml
+```
+
+You should now have 3 compose files which we can check with `ls`:
+
+```shell
+ls
+cht-core.yml  cht-couchdb.yml  docker-compose.yml
+```
+
+To start the first developer CHT instance, run `docker-compose`, prepending the needed environment variables:
+
+```shell script
+CHT_COMPOSE_PROJECT_NAME=app-devl COUCHDB_SECRET=foo DOCKER_CONFIG_PATH=${PWD} COUCHDB_DATA=${PWD}/couchd CHT_COMPOSE_PATH=${PWD} COUCHDB_USER=medic COUCHDB_PASSWORD=password docker-compose up
+```
+
+This may take some minutes to fully start depending on the speed of the Internet connection and speed of the bare-metal host. This is because docker needs to download all the storage layers for all the containers and the CHT needs to run the first run set up. After downloads and setup has completed, the CHT should be accessible on [https://localhost](https://localhost). You can log in with username `medic` and password `password`.
```

Comment:
```suggestion
This may take some minutes to fully start depending on the speed of the internet connection and speed of the host. This is because docker needs to download all the storage layers for all the containers and the CHT needs to run the first run set up. After downloads and setup has completed, the CHT should be accessible on [https://localhost](https://localhost). You can log in with username `medic` and password `password`.
```